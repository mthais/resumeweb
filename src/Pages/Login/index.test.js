import React from 'react'
import { MemoryRouter } from 'react-router'
import { mount } from 'enzyme'
import { LoginPage } from '.'

describe('LoginPage', () => {
  let wrapper
  let props

  beforeEach(() => {
    props = {
      isFetchingLoggingIn: false,
      handleLogin: jest.fn()
    }

    wrapper = mount(
      <MemoryRouter initialEntries={ [ '/' ] }>
        <LoginPage { ...props } />
      </MemoryRouter>
    )
  })

  it('should match with snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('when submit is called', () => {
    const data = { email: 'user@test.com', password: '123' }

    beforeEach(async () => {
      await wrapper.find(LoginPage).instance().submit(data)
    })

    it('should calls handleSubmit function with data', () => {
      expect(props.handleLogin).toBeCalledWith(data)
    })
  })
})
