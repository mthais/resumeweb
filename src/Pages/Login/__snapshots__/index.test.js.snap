// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match with snapshot 1`] = `"<div class=\\"col\\"><h2 class=\\"text-center page-title\\">Bem vindo de volta!</h2><p class=\\"text-wrapper\\">Para se manter conectado conosco, favor inserir seu endereço de Email e sua Senha</p><div class=\\"row\\"><div class=\\"col\\"><button class=\\"py-0 pt-1 btn btn-primary btn-block\\"><div class=\\"no-gutters row\\"><div class=\\"text-right col\\"><ion-icon style=\\"font-size: 28px;\\" name=\\"logo-linkedin\\"></ion-icon></div><div class=\\"text-left pt-1 col\\">Linked in</div></div></button></div></div><div class=\\"divider\\"><span>OU</span></div><form><div class=\\"position-relative form-group\\"><label for=\\"email\\" class=\\"mb-0\\">Email</label><div><input name=\\"email\\" id=\\"email\\" placeholder=\\"email@exemplo.com\\" type=\\"email\\" class=\\"form-control form-control\\" value=\\"\\"><div class=\\"invalid-feedback\\"></div></div></div><div class=\\"position-relative row form-group\\"><div class=\\"col\\"><label for=\\"password\\" class=\\"mb-0\\">Senha</label><div><input name=\\"password\\" id=\\"password\\" type=\\"password\\" class=\\"form-control form-control\\" value=\\"\\"><div class=\\"invalid-feedback\\"></div></div><a class=\\"link float-right\\" href=\\"/recoverypassword\\">Esqueceu sua senha</a></div></div><div class=\\"row\\"><div class=\\"col\\"><a class=\\"link\\" href=\\"/signup\\">Eu não tenho uma conta!</a></div><div class=\\"text-right col\\"><button type=\\"submit\\" disabled=\\"\\" class=\\"px-4 btn btn-primary disabled\\">Entrar</button></div></div></form></div>"`;
